### PR TITLE
make billing budget filter services updatable

### PR DESCRIPTION
**Pull request by shuyama1**
This resolves https://github.com/hashicorp/terraform-provider-google/issues/12185

If this PR is for Terraform, I acknowledge that I have:

- [ ] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [ ] [Generated Terraform](https://github.com/GoogleCloudPlatform/magic-modules#generating-the-terraform-providers), and ran `make test` and `make lint` to ensure it passes unit and linter tests.
- [ ] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [ ] [Ran](https://github.com/hashicorp/terraform-provider-google/blob/main/.github/CONTRIBUTING.md#tests) relevant acceptance tests (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [ ] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:bug
billingbudget: fixed a bug where `google_billing_budget.budget_filter.services` was not updating.
```

### CODE REVIEW

File: mmv1/third_party/terraform/tests/resource_billing_budget_test.go [hunk 1]
```diff
@@ -241,6 +241,7 @@ resource "google_billing_budget" "budget" {
     labels  = {
       label1 = "bar2"
     }
+	services = ["services/24E6-581D-38E5"] # Bigquery
```
Comment (melinath) [inline mmv1/third_party/terraform/tests/resource_billing_budget_test.go:244]: It looks like the indentation may be off? Probably doesn't break anything but would be nice to fix.

**Review by melinath — APPROVED** [review]
lgtm

File: mmv1/third_party/terraform/tests/resource_billing_budget_test.go (hunk 1 above)
Comment (shuyama1) [reply mmv1/third_party/terraform/tests/resource_billing_budget_test.go]: Fixed! Thanks for catching it!

**Pull request merged**